ntoskrnl/ke/freeze.c

```diff
@@ -49,7 +80,27 @@ KeFreezeExecution(IN PKTRAP_FRAME TrapFrame,
 #endif
 
 #ifdef CONFIG_SMP
-    // TODO: Add SMP support.
+    Prcb = KeGetCurrentPrcb();
+    TargetAffinity = KeActiveProcessors;
+    TargetAffinity &= ~Prcb->SetMember;
+    if (TargetAffinity)
+    {
+        for (i = 0, Current = 1; i < KeNumberProcessors; i++, Current <<= 1)
+        {
+            if (TargetAffinity & Current)
+            {
+                /* stop target processor */
+                KiIpiSend(Current, IPI_FREEZE);
```

Comment:
Why not send all IPIs at once?

Reply:
To verify each processor updated its field before moving on.

But you asking this has made me realize if I want to verify that I can just parse through the list of processors and check instead of sending one IPI at a time and checking.